zjit/src/hir_type/mod.rs

```diff
@@ -145,9 +146,13 @@ fn is_range_exact(val: VALUE) -> bool {
     val.class_of() == unsafe { rb_cRange }
 }
 
-fn is_class_exact(val: VALUE) -> bool {
-    // Objects with RUBY_T_CLASS type and not instances of Module
-    val.builtin_type() == RUBY_T_CLASS && val.class_of() != unsafe { rb_cModule }
+fn is_module_exact(val: VALUE) -> bool {
+    if val.builtin_type() != RUBY_T_MODULE {
+        return false;
+    }
+
+    let klass = unsafe { rb_obj_class(val) };
```

Comment:
We should add a comment about how this is different from `class_of` because it looks so similar but does a different thing